### PR TITLE
fix(tests): verify gradients through distCoeff and K in distort_point…

### LINKED ISSUE #3684: distort_points gradcheck only verifies gradients through points — distCoeff and K paths untested
### 🚀 Feature Description

The existing `test_gradcheck` in `tests/geometry/calibration/test_distort.py` 
passes `requires_grad=True` only to `points`. The gradient paths through 
`distCoeff` and `K` have never been formally verified.

This PR adds three targeted fixes:
1. Updates `test_gradcheck` to enable `requires_grad=True` on all differentiable inputs
2. Adds `test_gradcheck_distcoeff_only` — isolates the distortion coefficient gradient path
3. Adds `test_gradcheck_K_only` — isolates the camera intrinsic matrix gradient path

No source changes. The implementation is already correctly differentiable — 
this is purely missing test coverage.

### 📂 Feature Category

Testing Infrastructure

### 💡 Motivation

The core value of a differentiable distortion model is being able to optimise 
distortion coefficients via gradient descent — NeRF camera parameter tuning, 
differentiable bundle adjustment, joint calibration during training.

The existing gradcheck only tested d(output)/d(points). It never tested:
- d(output)/d(distCoeff)  ← required for intrinsic optimisation
- d(output)/d(K)          ← required for joint K + distortion optimisation

Without these, the test suite gives false confidence that distort_points is 
fully differentiable through all its parameters. A user who attempts to 
optimise distCoeff via gradient descent has no test coverage backing that use-case.

### 💭 Proposed Solution

Modify `tests/geometry/calibration/test_distort.py`:

# BEFORE (only points has grad):
points   = torch.rand(1, 8, 2, dtype=torch.float64, requires_grad=True)
K        = torch.rand(1, 3, 3, dtype=torch.float64)                     # ← no grad
new_K    = torch.rand(1, 3, 3, dtype=torch.float64)                     # ← no grad
distCoeff = torch.rand(1, 4, dtype=torch.float64)                       # ← no grad

# AFTER (all differentiable inputs have grad):
points    = torch.rand(1, 8, 2, dtype=torch.float64, requires_grad=True)
K         = torch.rand(1, 3, 3, dtype=torch.float64, requires_grad=True)
new_K     = torch.rand(1, 3, 3, dtype=torch.float64, requires_grad=True)
distCoeff = torch.rand(1, 4, dtype=torch.float64, requires_grad=True)

Plus two new focused tests:
- test_gradcheck_distcoeff_only: fixes points/K, differentiates only through distCoeff
- test_gradcheck_K_only: fixes points/distCoeff, differentiates only through K

All 11 tests pass locally (CPU, float64, verified with pytest).

### 🔄 Alternatives Considered

Leaving the tests as-is. Rejected because it provides false confidence — 
the gradcheck passes but it is not testing the inputs that matter most 
for the differentiable calibration use-case.

### 🎯 Use Cases

1. NeRF training: optimise camera distortion coefficients jointly with the 
   scene representation via gradient descent. Requires d(loss)/d(distCoeff).

2. Differentiable bundle adjustment: minimise reprojection error end-to-end, 
   backpropagating through the distortion model into K and distCoeff.

3. Sensor fusion: train camera calibration parameters jointly with a 
   downstream model. Requires full gradient flow through all camera parameters.

### 📝 Additional Context

Verified locally:

pytest tests/geometry/calibration/test_distort.py -v -k "gradcheck"

PASSED test_gradcheck[cpu]
PASSED test_gradcheck_distcoeff_only[cpu]
PASSED test_gradcheck_K_only[cpu]
3 passed in 0.47s

Full suite: 11 passed, 0 failed.

### 🤝 Contribution Intent

- [x] I plan to submit a PR to implement this feature
- [ ] I'm requesting this feature but not planning to implement it

### PR DESCRIPTION
**Pull request by AdityaX18**
## 📝 Description
Fixes  #3684

The existing `test_gradcheck` passes `requires_grad=True` only to `points`, 
leaving gradient paths through `distCoeff` and `K` completely unverified. 
This is the primary use-case gap: optimising distortion coefficients via 
gradient descent (NeRF camera tuning, bundle adjustment) requires 
`d(output)/d(distCoeff)` to be correct — and no test was checking it.

No source changes. The implementation was already correctly differentiable. 
This is purely missing test coverage.

---

## 🛠️ Changes Made
- [x] Updated `test_gradcheck` to enable `requires_grad=True` on all differentiable inputs (`points`, `K`, `new_K`, `distCoeff`)
- [x] Added `test_gradcheck_distcoeff_only` — isolates and verifies the distortion coefficient gradient path independently
- [x] Added `test_gradcheck_K_only` — isolates and verifies the camera intrinsic matrix gradient path independently

---

## 🧪 How Was This Tested?
- [x] **Unit Tests:** 3 gradcheck tests added, all passing. Full suite: 11/11 passed.
- [x] **Manual Verification:** pytest tests/geometry/calibration/test_distort.py -v -k "gradcheck" , 'PASSED test_gradcheck[cpu]' , 'PASSED test_gradcheck_distcoeff_only[cpu]' , 'PASSED test_gradcheck_K_only[cpu]' , '3 passed in 0.47spytest tests/geometry/calibration/test_distort.py -v' , ' 11 passed in 0.49s,
- [x] **Edge Cases:** gradcheck uses `fast_mode=True`, float64, eps=1e-6 — consistent with existing test conventions in this file.

---

## 🕵️ AI Usage Disclosure
- [x] 🟡 **AI-assisted:** I used AI for boilerplate/refactoring but have manually reviewed and tested every line.

---

## 🚦 Checklist
- [ ] I am assigned to the linked issue (required before PR submission)
- [ ] The linked issue has been approved by a maintainer
- [x] I have performed a **self-review** of my code (no "ghost" variables or hallucinations).
- [x] My code follows the existing style guidelines of this project.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.

---

## 💭 Additional Context
I am a GSoC 2026 applicant proposing to extend `kornia.geometry.calibration` 
with fully differentiable lens distortion models (Brown-Conrady + Kannala-Brandt 
forward projection). Proper gradcheck coverage of all inputs in `distort_points` 
is foundational to that work — this fix ensures the existing function is 
verifiably differentiable through every parameter before I build on top of it.

### CODE REVIEW

**Comment by AdityaX18** [comment]
@shijianjian @edgarriba can you review this? its been 2-3 weeks.

**Review by edgarriba — APPROVED** [review]
